### PR TITLE
Moments: multithread all the pixelwise operations

### PR DESCRIPTION
**Pull request by mdoube**
Single-threaded methods are really slow for pixelwise operations on large images. This commit makes Moments of Inertia calculation much faster (depending on hardware resources).

### CODE REVIEW

**Comment by mdoube** [comment]
Need to get up to date with bonej-org/master first

**Comment by mdoube** [comment]
OK, rebased instead of merged.

File: Legacy/bonej/src/main/java/org/bonej/plugins/Moments.java [hunk 1]
```diff
@@ -613,49 +674,67 @@ private static double getDensityFactor(final ImagePlus imp) {
...
+		final int d = imp.getStackSize();
+		double[] sliceXTmax = new double[d + 1];
+		double[] sliceYTmax = new double[d + 1];
+		double[] sliceZTmax = new double[d + 1];
+
+		final AtomicInteger ai = new AtomicInteger(startSlice);
+		final Thread[] threads = Multithreader.newThreads();
+		for (int thread = 0; thread < threads.length; thread++) {
+			threads[thread] = new Thread(() -> {
+				for (int z = ai.getAndIncrement(); z <= endSlice; z = ai.getAndIncrement()) {
```
Comment (rimadoma) [inline Legacy/bonej/src/main/java/org/bonej/plugins/Moments.java]: I would find
```
final int nThreads = threads.length;
for (int z = startSlice + thread; z <= endSlice; z += nThreads)
```
easier to understand, but that's just personal preference.

**Comment by alessandrofelder** [comment]
Clearly didn't rectify this :/ there is an unnecessary extra commit by me. Not entirely sure how to remove it. Easiest to merge and leave it there? @mdoube, do you want the honour of merging?

**Pull request merged**